### PR TITLE
New Package: clex

### PR DESCRIPTION
**Pull request by ThatGeekyWeeb**
In response to #23362

### CODE REVIEW

**Comment by jnbr** [comment]
This needs some cleanup. For a start, remove all those unrelated commits and force push again, so the pull request contains only the commit with message "New package: clex".

File: srcpkgs/clex/template [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Template file for 'clex'
+pkgname=clex
+version=4.6.patch9
+revision=1
+archs="x86_64 i868 x86_64-musl"
```
Comment (jnbr) [inline srcpkgs/clex/template:5]: Why is it restricted to these architectures? If there is no good reason, remove the line.

File: srcpkgs/clex/template [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+build_style="gnu-configure"
+configure_args="--prefix=/usr --sysconfdir=/etc --infodir=/usr/share/info --mandir=/usr/share/man --localstatedir=/var"
```
Comment (jnbr) [inline srcpkgs/clex/template:7]: Those are default values in void and are already set by the build-style, you can remove the entire line.

File: srcpkgs/clex/template [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+hostmakedepends="ncurses-devel ncurses"
```
Comment (jnbr) [inline srcpkgs/clex/template:8]: This is most likely wrong and should instead go into makedepends. See Manual.md for difference between hostmakedepends and makedepends.

File: srcpkgs/clex/template [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+short_desc="File manager with a full-screen user interface"
+maintainer="ThatGeekyWeeb <thatgeekyweeb@gmail.com>"
+license="GPL-2.0-or-later"
+homepage="http://www.clex.sk/"
+distfiles="http://www.clex.sk/download/clex-${version}.tar.gz"
+checksum="4c866d9d7bb4e276f1c54fb897d71f1d6390ec1535ffb174798a8334d1f545e2"
+do_build() {
+	./configure ${configure_args}
```
Comment (jnbr) [inline srcpkgs/clex/template:16]: configure is already called by the build-style in the do_configure step, remove it here

File: srcpkgs/clex/template [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	make
+	yes | ./src/cfg-clex
+}
```
Comment (jnbr) [inline srcpkgs/clex/template:19]: The build-style already calls make, there is no need to define a do_build function.
If you need to run some non-standard things like cfg-clex, you can do it in post_build

File: srcpkgs/clex/template [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+do_install() {
+	make INSTALL_ROOT=${DESTDIR} install
+	vbin ./src/clex
+}
```
Comment (jnbr) [inline srcpkgs/clex/template:23]: same as with so_build

File: srcpkgs/clex/template (hunk 1 above)
Comment (ThatGeekyWeeb) [reply srcpkgs/clex/template:5]: I restricted it as it fails to pass the checks on arm based arches, due to 
> `CLEX requires a CURSES library with a wide character support`

I am currently unsure which dependency I need to add to fix said error, I'll work on fixing that.

**Comment by ThatGeekyWeeb** [comment]
I'm having a re-occurring issue with packaging `clex`
It depends `ncurses-devel` which depends `ncurses-libs` 
`ncurses` also depends `ncurses-libs`
After `xbps-src` installs `ncurses-devel` it tries to remove the autodep of `ncurses-libs` but that conflicts `ncurses`
Due to this error, I can't continue to pkg `clex`
```bash
=> clex-4.6.patch9_1: building [gnu-configure] for x86_64...
   [target] ncurses-devel-6.2_2: found (/host/binpkgs/clex)
   [runtime] ncurses-devel-6.2_2: found (/host/binpkgs/clex)
=> clex-4.6.patch9_1: installing target dependencies: ncurses-devel-6.2_2 ...
=> clex-4.6.patch9_1: removing autodeps, please wait...
=> ERROR: clex-4.6.patch9_1: failed to install 'ncurses-devel-6.2_2' dependency! (error 19)
ncurses-libs-6.2_2 in transaction breaks installed pkg `ncurses-6.2_3'
Transaction aborted due to unresolved dependencies.
=> ERROR: Please see above for the real error, exiting...
```

**Comment by ericonr** [comment]
@ThatGeekyWeeb you need to update your repository with the latest packages. Add this repo as an upstream and rebase your commits onto it.

**Comment by ThatGeekyWeeb** [comment]
Thx @ ericonr, that fixed my issue! :)

File: srcpkgs/clex/template [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Template file for 'clex'
+pkgname=clex
+version=4.6.patch9
+revision=1
+build_style="gnu-configure"
+makedepends="ncurses-devel ncurses-devel-32bit"
```
Comment (ericonr) [inline srcpkgs/clex/template:6]: The 32bit version isn't necessary.

**Comment by ThatGeekyWeeb** [comment]
> Why is it restricted to these architectures? If there is no good reason, remove the line.
> I restricted it as it fails to pass the checks on arm based arches, due to

> `    CLEX requires a CURSES library with a wide character support`

> I am currently unsure which dependency I need to add to fix said error, I'll work on fixing that.


I can confirm that arm arches now build fine!
> `makedepends="ncurses-devel ncurses-libtinfo-devel"` 

works!

**Pull request closed without merging**